mlir/test/Conversion/MemRefToEmitC/memref-to-emitc.mlir

```diff
@@ -94,3 +94,12 @@ func.func @memref_collapse_shape(%arg: memref<2x5xi32>) -> memref<10xi32> {
   %0 = memref.collapse_shape %arg [[0, 1]] : memref<2x5xi32> into memref<10xi32>
   return %0 : memref<10xi32>
 }
+
+// -----
+
+// CHECK-LABEL: memref_reinterpret_cast
+func.func @memref_reinterpret_cast(%arg: memref<2x5xi32>) -> memref<10xi32> {
+  // CHECK: emitc.cast %{{[^ ]*}} : !emitc.array<2x5xi32> to !emitc.array<10xi32> ref
+  %0 = memref.reinterpret_cast %arg to offset: [0], sizes: [10], strides: [1] : memref<2x5xi32> to memref<10xi32>
```

Comment:
I would rather fold this reinterpret_cast into an equvialent collapse_shape. then lowering it. For this PR, can you come up with an example reinterpret_cast  that isn't a collapse/expand shape?